lib/src/test/java/tech/relaycorp/relaydroid/FirstPartyEndpointTest.kt

```diff
@@ -0,0 +1,68 @@
+package tech.relaycorp.relaydroid
+
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.argumentCaptor
+import com.nhaarman.mockitokotlin2.eq
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNull
+import org.junit.Test
+import tech.relaycorp.relaydroid.test.FirstPartyEndpointFactory
+import tech.relaycorp.relaydroid.test.ParcelDeliveryCertificates
+import tech.relaycorp.relaynet.testing.pki.KeyPairSet
+import tech.relaycorp.relaynet.wrappers.privateAddress
+import java.security.KeyPair
+
+class FirstPartyEndpointTest {
+
+    @Test
+    fun address() {
+        val endpoint = FirstPartyEndpointFactory.build()
+        assertEquals(endpoint.keyPair.public.privateAddress, endpoint.address)
+    }
+
+    @Test
+    fun register() = runBlockingTest {
+        val gateway = mock<GatewayClientI>()
+        val storage = mock<StorageImpl>()
+        Relaynet.storage = storage
+        Relaynet.gatewayClientImpl = gateway
```

Comment:
I think this would be good to move to a `setUp()` method to avoid noise in the tests themselves.